lib/vanagon/platform/windows.rb

```diff
@@ -53,9 +53,40 @@ def generate_packaging_artifacts(workdir, name, binding)
       # @param name [String] name of the project
       # @param binding [Binding] binding to use in evaluating the packaging templates
       def generate_msi_packaging_artifacts(workdir, name, binding)
-        FileUtils.mkdir_p(File.join(workdir, "wix"))
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.wxs.erb"), File.join(workdir, "wix",  "#{name}.wxs"), false, { :binding => binding })
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.filter.xslt.erb"), File.join(workdir, "wix", "#{name}.filter.xslt"), false, { :binding => binding })
+        # Recurse through the list of files in the resources/windows/wix directory and copy them over to the work directory.
+        copy_wix_artifacts_to_workdir("resources/windows/wix", workdir, name, binding)
+        copy_wix_artifacts_to_workdir(File.join(VANAGON_ROOT, "resources/windows/wix"), workdir, name, binding)
```

Comment:
Specifically the exists clause on line 83 only realistically executes on the second run of the function which is a little confusing and unreadable. What would be better would be to simply glob the project files first, then glob any default vanagon wix files that aren't there yet (whcih would be the exists clause) and then execute erb on any .erb files in the workdir. that way you could run the function only once and the whole thing is more readable when we need to change things